### PR TITLE
Update seeders w/ campaigns & remove Contentful ID functionality.

### PR DESCRIPTION
**Pull request by DFurnes**
#### What's this PR do?
This pull request removes the ability to send a Contentful ID as `campaign_id`, and updates our database seeders to create and reference local campaign IDs for signups/posts:

🌱 The seeder will now create ten campaigns, and attach signups/posts to them (rather than just letting each signup or post create it's own campaign ID).

🏕 If a `campaign_id` is not provided, the model factory will now create a corresponding campaign resource for any posts or signups.

🏹 Fixes an issue where a factory would create a post and the auto-generated signup for it would have a different user & campaign ID.

🕰 By default, ensure that seeded campaigns were created in the past and end in the future (previously could have been created anytime in the past & would have ended 3 months from then).

🛂 Removes support for passing Contentful IDs into `campaign_id`, since we're no longer planning on using this as our unique campaign ID (and so we'd just be getting ourselves invalid data by accepting these).

#### How should this be reviewed?
Tests should pass!

#### Any background context you want to provide?
I'm working on a follow-up PR (for Rogue Admin) that relies on these new seeds. Stay tuned!

#### Relevant tickets
References [#161650678](https://www.pivotaltracker.com/story/show/161650678).

#### How to deploy
https://github.com/DoSomething/rogue/blob/master/docs/development/deployments.md

### CODE REVIEW

File: database/factories/ModelFactory.php [hunk 1]
```diff
@@ -99,30 +100,23 @@
...
-        'id' => $faker->numberBetween($min = 9000, $max = 20000),
-        'internal_title' => $faker->sentence(),
-        'cause' => $causes[array_rand($causes)],
+        'internal_title' => title_case($faker->unique()->catchPhrase),
+        'cause' => $faker->randomElement([
+            'Animals', 'Bullying', 'Disasters', 'Discrimination', 'Education',
+            'Environment', 'Homelessness', 'Mental Health', 'Physical Health',
+            'Poverty', 'Relationships', 'Sex', 'Violence',
+        ]),
         'impact_doc' => 'https://www.google.com/',
```
Comment (chloealee) [inline database/factories/ModelFactory.php]: ah love this!

File: database/seeds/DatabaseSeeder.php [hunk 2]
```diff
@@ -14,43 +14,49 @@ class DatabaseSeeder extends Seeder
...
 
-        // Create 10 signups with ony accepted posts, from lil' angels!
-        factory(Signup::class, 10)->create(['campaign_id' => 46])->each(function (Signup $signup) {
-            $signup->posts()->save(factory(Post::class, 'accepted')->create([
-                'signup_id' => $signup->id,
-                'campaign_id' => $signup->campaign_id,
-                'northstar_id' => $signup->northstar_id,
-            ]));
-        });
+            // Create 5-10 signups with only accepted posts, from lil' angels!
```
Comment (chloealee) [inline database/seeds/DatabaseSeeder.php]: 👼

File: database/seeds/DatabaseSeeder.php [hunk 3]
```diff
@@ -14,43 +14,49 @@ class DatabaseSeeder extends Seeder
...
 
-        // Create 10 signups with rejected posts, from troublemakers!
-        factory(Signup::class, 10)->create()->each(function (Signup $signup) {
-            $signup->posts()->save(factory(Post::class, 'rejected')->create([
-                'signup_id' => $signup->id,
-                'campaign_id' => $signup->campaign_id,
-                'northstar_id' => $signup->northstar_id,
-            ]));
-        });
+            // Create 5-10 signups with rejected posts, from troublemakers!
```
Comment (chloealee) [inline database/seeds/DatabaseSeeder.php]: 😈

File: docs/endpoints/posts.md [hunk 4]
```diff
@@ -162,8 +162,8 @@ This will automatically create or update the corresponding signup.
 POST /api/v3/posts
 ```
 
-- **campaign_id**: (int|string) required.
-  The Drupal/Contentful ID of the campaign that the user's post is associated with.
+- **campaign_id**: (int) required.
+  The Campaign ID of the campaign that the user's post is associated with.
 - **campaign_run_id**: (int)
```
[the remark this answers is not in the document]
Comment (DFurnes) [reply docs/endpoints/posts.md]: Yeah, I'm going to fully remove support for these in a follow-up PR after we've run the script on production (for now, you can still provide them in a couple scenarios).

File: database/seeds/DatabaseSeeder.php [hunk 5]
```diff
@@ -14,43 +14,49 @@ class DatabaseSeeder extends Seeder
...
-        factory(Signup::class, 20)->create()->each(function (Signup $signup) {
-            $signup->posts()->save(factory(Post::class, 'accepted')->create([
-                'signup_id' => $signup->id,
-                'campaign_id' => $signup->campaign_id,
-                'northstar_id' => $signup->northstar_id,
-            ]));
+        // Create 10 campaigns with signups & posts.
+        factory(Campaign::class, 10)->create()->each(function (Campaign $campaign) {
+            // Create 10-30 signups with one accepted post & some pending posts.
+            factory(Signup::class, rand(10, 20))->create(['campaign_id' => $campaign->id])
```
[the remark this answers is not in the document]
Comment (DFurnes) [reply database/seeds/DatabaseSeeder.php]: Oh, lemme fix up that comment. 10-20 is probably fine.

File: database/factories/ModelFactory.php [hunk 6]
```diff
@@ -99,30 +100,23 @@
...
-        'start_date' => $start_date,
-        // Make sure the end date is after the start date
-        'end_date' => date('m/d/Y', strtotime("+3 months", strtotime($start_date))),
+        // By default, we create an "open campaign".
+        'start_date' => $faker->dateTimeBetween('-6 months', 'now'),
+        'end_date' => $faker->dateTimeBetween('+1 months', '+6 months'),
```
[the remark this answers is not in the document]
Comment (DFurnes) [reply database/factories/ModelFactory.php]: That's a good idea!

[the remark this answers is not in the document]
Comment (katiecrane) [reply database/factories/ModelFactory.php]: definitely doesn't block merge though, just a thought!

File: database/seeds/DatabaseSeeder.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (DFurnes) [reply database/seeds/DatabaseSeeder.php]: Fixed in d2a66de.

**Pull request merged**